src/goto-programs/goto_program.h

```diff
@@ -287,6 +287,13 @@ class goto_programt
       return guard;
     }
 
+    /// Get the condition of gotos, assume, assert
+    exprt &get_condition()
```

Comment:
I believe we do have a lot of places where we have a const and a non-const variant with the same name, but I also think that in those cases we neither have a `get_` nor a `set_` prefix? What I would have used is `void set_condition(exprt cond) { PRECONDITION(has_condition()); guard = std::move(cond); }`, which of course requires a somewhat bigger change in that all users need to do `exprt modified_condition = get_condition(); do_something(modified_condition); set_condition(std::move(modified_condition));` But certainly nothing that I intend to block on, my approval remains in place.

Reply:
Indeed; I couldn't decide. To write to stuff thats returned by `get_X` seems to be very common in Java, but much less common in C++.
The key reason why I went for non-const get is that we have a lot of instances where we apply a transform to the guard. That would then become
`exprt tmp = instr.get_condition();
TRANSFORM(tmp);
instr.set_condition(tmp);`
That could possibly be changed into
`instr.set_condition(TRANSFORM(instr.get_condition());`
if we turn all those TRANSFORM into functional style.

Reply:
I'll action this.

Reply:
Another option would be to offer `instructiont::transform()`, which would take a lambda that modifies the expressions.